gluten-substrait/src/main/scala/org/apache/gluten/backendsapi/TransformerApi.scala

```diff
@@ -76,4 +76,7 @@ trait TransformerApi {
   def invalidateSQLExecutionResource(executionId: String): Unit = {}
 
   def genWriteParameters(fileFormat: FileFormat, writeOptions: Map[String, String]): Any
+
+  /** use Hadoop Path class to encode the file path */
+  def encodeFilePathIfNeed(filePath: String): String = filePath
```

Comment:
It turn out that a similar difference exists on regular scan

VL https://github.com/apache/incubator-gluten/blob/e56da333dc33de90540efa6dc63efa70c3c71ce8/backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxIteratorApi.scala#L124-L126
CH https://github.com/apache/incubator-gluten/blob/e56da333dc33de90540efa6dc63efa70c3c71ce8/backends-clickhouse/src/main/scala/org/apache/gluten/backendsapi/clickhouse/CHIteratorApi.scala#L166

Maybe the logics should be somehow consolidated in future, I am not sure.